File: A2/Sem3/MAP/Seminar/Seminar11/Seminar11/MessageTask.cs
abstract class Task
{
    public string taskID { get; set; }
    public string Descriere { get; set; }

    public Task(string taskID, string descriere)
    {
        this.taskID = taskID;
    }

    public abstract void execute();

    public override string ToString()
    {
        return taskID + " " + Descriere;
    }


    public override int GetHashCode()
    {
        return taskID.GetHashCode();
    }
}

class MessageTask : Task
{
    public string mesaj { get; set; }
    public string from { get; set; }
    public string to { get; set; }
    public DateTime Date { get; set; }

    public MessageTask(string taskid, string descriere, string mesaj, string from, string to, DateTime date) : base(taskid, descriere)
    {
        this.Date = date;
        this.from= from;
        this.to= to;
        this.mesaj = mesaj;
    }

    public override void execute()
    {
        Console.WriteLine(this.mesaj + "(" + this.Date.ToString("dd.MM.yyyy hh:mm") + ")");
    }
}

File: A2/Sem3/MAP/Laborator/MAPL2/src/model/MessageTask.java
package model;

import utils.Constants;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MessageTask extends Task {
    private String message;
    private String from;
    private String to;
    private LocalDateTime date;

    public MessageTask(String _taskID, String _description, String message, String from, String to, LocalDateTime date) {
        super(_taskID, _description);
        this.message = message;
        this.from = from;
        this.to = to;
        this.date = date;
    }

    @Override
    public void execute() {
        System.out.println(toString());
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    @Override
    public String toString()
    {
        return super.toString() + " " + getMessage() +
                "\nFrom: " + getFrom() + "\nData: " +
                getDate().format(Constants.DATE_TIME_FORMATTER);
    }
}
